pkg/controller/services.go

```diff
@@ -978,18 +1056,50 @@ func (cont *AciController) opflexDeviceDeleted(dn string) {
 }
 
 func (cont *AciController) writeApicSvc(key string, service *v1.Service) {
-	endpointsobj, _, err := cont.endpointsIndexer.GetByKey(key)
-	if err != nil {
-		cont.log.Error("Could not lookup endpoints for " +
-			key + ": " + err.Error())
-		return
-	}
-
 	aobj := apicapi.NewVmmInjectedSvc(cont.vmmDomainProvider(),
 		cont.config.AciVmmDomain, cont.config.AciVmmController,
 		service.Namespace, service.Name)
 	aobjDn := aobj.GetDn()
 	aobj.SetAttr("guid", string(service.UID))
+	if !cont.endPointSliceEnabled {
```

Comment:
Let's avoid this if check and use some abstraction so this code doesn't need to know the details of the k8s object.